ansible/roles/sap-hana-fast-restart/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Determine the number of CPU sockets available in the HANA server
+  shell: |
+    cat /sys/devices/system/node/node*/meminfo | grep MemTotal | awk 'BEGIN {printf "%10s | %20s\n", "NUMA NODE", "MEMORY GB"; while (i++ < 33) printf "-"; printf "\n"} {printf "%10d | %20.3f\n", $2, $4/1048576}'
+  register: cpu_sockets
+
+- name: Read global.ini entries
+  shell: |
+    cat {{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/global.ini
+  register: global_ini
+
+- name: Read indexserver.ini entries
+  shell: |
+    cat {{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/indexserver.ini
+  register: indexserver_ini
+
+- name: Create tmpfs directories for numa nodes
+  file:
+    path: "/hana/tmpfs{{ item }}/{{ sap_hana_sid }}"
+    state: directory
+    owner: "{{ sap_hana_user }}"
+    group: sapsys
+    mode: '777'
+    recurse: yes
+  with_sequence: start=0 end='{{ cpu_sockets.stdout_lines|length - 3 }}'
+  when: global_ini.stdout.find('basepath_persistent_memory_volumes') == -1
+
+- name: Mount and make fstab for tmpfs entries
+  mount:
+    name: "/hana/tmpfs{{ item }}/{{ sap_hana_sid }}"
+    fstype: tmpfs
+    opts: "rw,relatime,mpol=prefer:{{ item }}"
+    src: "tmpfs{{ sap_hana_sid }}{{ item }}"
+    state: mounted
+  with_sequence: start=0 end='{{ cpu_sockets.stdout_lines|length - 3 }}'
+  when: global_ini.stdout.find('basepath_persistent_memory_volumes') == -1
+
+- name: Add a line to a file to configure SAP HANA to use them with the right Numa Node correlation
+  lineinfile:
+    path: '{{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/global.ini'
+    insertafter: '^\[persistence\]'
+    line: 'basepath_persistent_memory_volumes = '
+    create: yes
+  when: global_ini.stdout.find('basepath_persistent_memory_volumes') == -1
+
+- name: Add NUMA nodes to the global.ini file
+  lineinfile:
+    path: '{{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/global.ini'
+    backrefs: true
+    regexp: '^(basepath_persistent_memory_volumes = .*)$'
+    line: '\1/hana/tmpfs{{ item }}/{{ sap_hana_sid }};'
+  with_sequence: start=0 end='{{ cpu_sockets.stdout_lines|length - 3 }}'
+  when: global_ini.stdout.find('basepath_persistent_memory_volumes') == -1
+
+- name: Update which column tables or partitions shall be stored into the persistent memory
+  lineinfile:
+    path: '{{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/indexserver.ini'
+    insertafter: EOF
+    line: '{{ item }}'
+    create: yes
+  with_items:
+    - '[persistent_memory]'
+    - 'table_default = ON'
+  when: indexserver_ini.stdout.find('persistent_memory') == -1
+
+- name: Include sap hana stop role
```

Comment:
The `include_role` tasks for `sap-hana-stop` and `sap-hana-start` would benefit from e.g. `when: global_ini is changed or indexserver_ini is changed` otherwise it will always be restarted which is slow.

Reply:
Done. Pushed the change.